atom/browser/native_window_mac.mm

```diff
@@ -382,6 +393,19 @@ - (void)drawRect:(NSRect)dirtyRect {
   // We will manage window's lifetime ourselves.
   [window_ setReleasedWhenClosed:NO];
 
+  // Configure title bar look on Yosemite or newer
+  if (base::mac::IsOSYosemiteOrLater()) {
+    if ((titleBarStyle == "hidden") || (titleBarStyle == "hidden-inset")) {
+      [window_ setTitlebarAppearsTransparent:YES];
+      [window_ setTitleVisibility:NSWindowTitleHidden];
+      if (titleBarStyle == "hidden-inset") {
+        NSToolbar *toolbar = [[NSToolbar alloc] initWithIdentifier:@"titlebarStylingToolbar"];
+        toolbar.showsBaselineSeparator = NO;
+        [window_ setToolbar:toolbar];
```

Comment:
My bad. Thanks for pointing this out. I assumed ARC, but apparently not, been a while since I worked on a non-ARC codebase. :)